yjit/src/backend/ir.rs

```diff
@@ -151,6 +151,15 @@ impl Opnd
         }
     }
 
+    pub fn sub_opnd(&self, num_bits: u8) -> Option<Opnd> {
+        match *self {
```

Comment:
We should probably add an assertion that num_bits is <= than the current num_bits and that it's 8, 16, 32 or 64.

Reply:
addressed this in https://github.com/ruby/ruby/pull/6359/commits/6d032ca60f31e8bc12a99dfbe504e22cef01af42 and https://github.com/ruby/ruby/pull/6359/commits/236b9800c7fd99c89780f36d4dc549be0b40db49.